scripts/vsphere/attach_vm

```diff
@@ -160,12 +162,95 @@ write_config_context_vm() {
 
   write_config_context
 
-
   echo "export VM_APP=${VM_APP}" >>${CONTEXT_FILE}
   echo "export VM_IP=${VM_IP}" >>${CONTEXT_FILE}
 
   rm ".attachConfig"
   cat ${OUTPUT_DIR}/vm-${VM_APP}/".context"
 }
 
+# multi nic support
+multinicdetection() {
+  # cache all VM ip table info for later use
+  IP_INFO=$(ssh -i "${KEY_FILE}" -o StrictHostKeyChecking=no root@${VM_IP} ip -j a )
+
+  NICCOUNTSTRING=$(echo "${IP_INFO}" | jq -jr '.[].ifname| type')
+  NICSTRING=$(echo "${IP_INFO}" | jq -jr '.[]|" ",.ifname')
+
+  # Case 1: VM has only one nic. fill in MANAGED_ADDR and MANAGED_NIC if unspecified.
+  if [[ ${#NICCOUNTSTRING} == 12 ]]; then
+    VMNICNAME=$(echo "${IP_INFO}"| jq -r '.[] | select(.ifname=="lo" | not) | .ifname')
+    VMNICIP=$(echo "${IP_INFO}"| jq -r '.[] | select(.ifname=="$(VMNICNAME) | .addr_info[] | select(.family == "inet") | select(.scope == "global") | .local')
+    if [[ -z "${MANAGED_NIC}" ]]; then
+      MANAGED_NIC="${VMNICNAME}"
+    elif [[ -z "${MANAGED_ADDR}" ]]; then
+      MANAGED_ADDR="${VMNICIP}"
+    fi
+
+    INPUTNICIP=$(echo "${IP_INFO}"| jq -r '.[] | select(.ifname=="$(MANAGED_ADDR) | .addr_info[] | select(.family == "inet") | select(.scope == "global") | .local')
+
+    if [[ "${MANAGED_ADDR}" != "${VMNICIP}" || "${MANAGED_NIC}" != "${VMNICNAME}" ]]; then
+      # User might input a wrong addr or nic which doesn't match MANAGED_ADDR or MANAGED_NIC filled in above.
+      fatal "The VM has only one nic. nic name: ${VMNICNAME}, nic address: ${VMNICIP}. But user is specifying some other nic name or nic address."
+    elif [[ "${INPUTNICIP}" != "${VMNICIP}" ]]; then
+      # User might input a unmatched addr and nic
+      fatal "The VM has only one nic. nic name: ${VMNICNAME}, nic address: ${VMNICIP}. But user input for nic name and nic address is not matched."
+    fi
+  fi
```

Comment:
Prefer `cannot find the nic ${MANAGED_NIC} or ${MANAGED_ADDR}`

Reply:
Is the point to not disclose that much info to user? Sure, changed.